### ISSUE TITLE
Add options to include/exclude paths

### ISSUE THREAD

**Issue by willfaught** [golangci/golangci-lint#4635]
### Your feature request related to a problem? Please describe

I use https://entgo.io. It generates an /ent directory, and expects you to put your schema definitions in /ent/schema. It then generates code in /ent/foo, /ent/bar, etc. I have no control over the generated code, so I don't want to see linter errors for it. I don't see a way to exclude /ent/\*\*, but include /ent/schema.

### Describe the solution you'd like

run --exclude-path '^ent/.\*$' --include-path '^ent/schema/.\*$'

### Describe alternatives you've considered

--exclude-dirs only works on directory names, which won't work if you want to skip some directories and exclude others with the same name.

**Comment by ldez** [comment]
Hello,

you can use `exclude-files`: https://golangci-lint.run/usage/false-positives/#exclude-or-skip

**Comment by willfaught** [comment]
I don't see how exclude-files can exclude /ent/**, but include /ent/schema. Adjusting my example above, `run --exclude-files '^ent/.*$'` will exclude ent/schema.

**Comment by denis-tingaikin** [comment]
I see a few options for how we can manage it.

1. Add a new flag that will exclude things from the exluded_path, as suggested by @willfaught. Here we can simplify the re-use approach from the golangci-linter/issues config, where we have exlude_files and include options.
2. Allow using regexp in `exlude_files` with negative lookahead. E.g., use this lib: https://github.com/dlclark/regexp2
3.  List all paths in `--exclude-path`  except `'^ent/schema/.*$'`. For example,  `--exclude-path ent/folder1, ent/folder2/.*, ent/folder3/.*` and etc 😀

@ldez Thoughts?

**Comment by ldez** [comment]
This can be achieved like that:

```yml
issues:
  exclude-rules:
    - path-except: ^ent/schema/.*$
      text: .+

```
